src/Connector.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.18;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {ICurvePool} from "./interfaces/ICurvePool.sol";
+import {IStrategyInterface} from "./interfaces/IStrategyInterface.sol";
+
+contract Connector {
```

Comment:
is this meant to implement a certain interface/standard or it's all up to us how we build this?

Reply:
i like the approach with this connector instead of overloading the strategy with more types of deposit functions

in the end the vault is still going to hold the strategy LPs and needs a rate on those but seems cleaner to get them or redeem them like this